app/concerns/gkr_keyword_recommendations.rb

```diff
@@ -7,14 +7,24 @@ def fetch_keyword_recommendations(user, provider)
     response = cmr_client.fetch_keyword_recommendations(abstract)
 
     if response.error?
+      if response.status >= 500 && response.status <= 505    # NGAP LB can't connect to GKR.
+        log_gkr_comm_error(user, provider, abstract, response.status, 'Communication failure with GKR')
+        return { 'error': 'CommError' }
+      end
+
       log_gkr_request_failed(user, provider, abstract, response)
-      { id: '', recommendations: [] }
+      return { id: '', recommendations: [] }
     end
 
     keyword_recommendations = response.body['recommendations'].map { |rec| rec['keyword'] }
     { id: response.body['uuid'], recommendations: keyword_recommendations }
   end
 
+  def log_gkr_comm_error(user, provider, abstract, status, reason)
```

Comment:
Can we add a request id # to this?  So we can match up these log statements to the user's error displayed

Reply:
See ProviderOrdersController.rb, it looks like they just use the request object to grab the uuid,  i.e.
    flash[:alert] = "The order request #{request.uuid} timed out showing the order. Please try again or contact #{view_context.mail_to('support@earthdata.nasa.gov', 'Earthdata Support')}."

Reply:
Currently, the GKR API creates a request id as it returns the keywords.   In this case,   we haven't yet communicated with the API, so we don't know what the request id will be.   In the successful cases, the log messages will log the request id, and it will be stored in the db.

All of this violates our security control which says the client will create the request id and send it through a standard header.  I believe we have a separate ticket to fix this, but it will require changes by the impact team as well as  mmt.